tests/api_connexion/endpoints/test_extra_link_endpoint.py

```diff
@@ -15,24 +15,198 @@
     @classmethod
     def setUpClass(cls) -> None:
         super().setUpClass()
-        cls.app = app.create_app(testing=True)  # type:ignore
+        with mock.patch.dict('os.environ', SKIP_DAGS_PARSING='True'):
+            cls.app = app.create_app(testing=True)  # type:ignore
+
+    @provide_session
+    def setUp(self, session) -> None:
+        self.now = datetime(2020, 1, 1)
+
+        clear_db_runs()
+        clear_db_xcom()
+
+        self.dag = self._create_dag()
+        self.app.dag_bag.dags = {self.dag.dag_id: self.dag}  # type: ignore  # pylint: disable=no-member
+        self.app.dag_bag.sync_to_db()   # type: ignore  # pylint: disable=no-member
+
+        dr = DagRun(
+            dag_id=self.dag.dag_id,
+            run_id="TEST_DAG_RUN_ID",
+            execution_date=self.now,
+            run_type=DagRunType.MANUAL.value,
+        )
+        session.add(dr)
+        session.commit()
 
-    def setUp(self) -> None:
         self.client = self.app.test_client()  # type:ignore
 
-    @pytest.mark.skip(reason="Not implemented yet")
+    def tearDown(self) -> None:
+        super().tearDown()
+        clear_db_runs()
+        clear_db_xcom()
+
+    @staticmethod
+    def _create_dag():
+        with DAG(dag_id="TEST_DAG_ID", default_args=dict(start_date=days_ago(2),)) as dag:
+            BigQueryExecuteQueryOperator(task_id="TEST_SINGLE_QUERY", sql="SELECT 1")
+            BigQueryExecuteQueryOperator(task_id="TEST_MULTIPLE_QUERY", sql=["SELECT 1", "SELECT 2"])
+        return dag
+
+    @parameterized.expand(
+        [
+            (
+                "/api/v1/dags/INVALID/dagRuns/TEST_DAG_RUN_ID/taskInstances/TEST_SINGLE_QUERY/links",
+                'DAG not found'
+            ),
+            (
+                "/api/v1/dags/TEST_DAG_ID/dagRuns/INVALID/taskInstances/TEST_SINGLE_QUERY/links",
+                "DAG Run not found"
+            ),
+            (
+                "/api/v1/dags/TEST_DAG_ID/dagRuns/TEST_DAG_RUN_ID/taskInstances/INVALID/links",
+                "Task not found"
+            ),
+        ]
+    )
+    def test_should_response_404_on_invalid_task_id(self, url, expected_title):
+        response = self.client.get(url)
+
+        self.assertEqual(404, response.status_code)
+        self.assertEqual({'detail': None, 'status': 404, 'title': expected_title, 'type': 'about:blank'}, response.json)
+
+    @mock_plugin_manager(plugins=[])
     def test_should_response_200(self):
+        XCom.set(
+            key="job_id",
+            value="TEST_JOB_ID",
+            execution_date=self.now,
+            task_id="TEST_SINGLE_QUERY",
+            dag_id=self.dag.dag_id,
+        )
         response = self.client.get(
-            "/dags/TEST_DG_ID/dagRuns/TEST_DAG_RUN_ID/taskInstances/TEST_TASK_ID/links"
+            "/api/v1/dags/TEST_DAG_ID/dagRuns/TEST_DAG_RUN_ID/taskInstances/TEST_SINGLE_QUERY/links"
+        )
+
+        self.assertEqual(200, response.status_code, response.data)
```

Comment:
```suggestion
        self.assertEqual(200, response.status_code)
```
I have never used it but I guess response.data returns the response.json in bytes literal?

Reply:
3 argument is just a message for the developer. It can be in the form of bytes if it is safer and more readable.